src/Raven.Server/Documents/Handlers/QueriesHandler.cs

```diff
@@ -47,44 +47,43 @@ public Task Get()
 
         public async Task HandleQuery(HttpMethod httpMethod)
         {
-            using (var tracker = new RequestTimeTracker(HttpContext, Logger, Database, "Query"))
+            RequestTimeTracker tracker = null;
+            try
             {
-                try
+                using (var token = CreateHttpRequestBoundTimeLimitedOperationTokenForQuery())
+                using (var queryContext = QueryOperationContext.Allocate(Database))
+                using (tracker = new RequestTimeTracker(HttpContext, Logger, Database, "Query"))
```

Comment:
we need to revert the code and implement double-disposal prevention mechanism and dispose manually after `await Debug()` and `await Query`

Reply:
Added